### ISSUE TITLE
Cannot migrate to v13.2

### ISSUE THREAD

**Issue by mehmet-erim** [nrwl/nx#7866]
I tried to migrate [our repo](https://github.com/abpframework/abp/tree/dev/npm/ng-packs) from v13.1 to v13.2 but I faced the error below:

```
Running migration migration-v13-testbed-teardown
Successfully finished migration-v13-testbed-teardown
---------------------------------------------------------
Running migration migration-v13
    Could not find TypeScript project for project: account
    Could not find TypeScript project for project: account-core
    Could not find TypeScript project for project: components
    Could not find TypeScript project for project: core
directoryExists is not a function
/Users/user1/Documents/abp/npm/ng-packs/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/yargs.js:1132
                throw err;
                ^

Error: Command failed: /var/folders/_b/1_14x_0j78n6n73jz8w5sw740000gq/T/tmp-47134-STC0t0TWw895/node_modules/.bin/tao migrate --run-migrations
    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at Object.handler (/Users/user1/Documents/abp/npm/ng-packs/node_modules/@nrwl/workspace/src/command-line/nx-commands.js:125:42)
    at Object.runCommand (/Users/user1/Documents/abp/npm/ng-packs/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/command.js:196:48)
    at Object.parseArgs [as _parseArgs] (/Users/user1/Documents/abp/npm/ng-packs/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/yargs.js:1043:55)
    at Object.get [as argv] (/Users/user1/Documents/abp/npm/ng-packs/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/yargs.js:986:25)
    at initLocal (/Users/user1/Documents/abp/npm/ng-packs/node_modules/@nrwl/cli/lib/init-local.js:28:79)
    at Object.<anonymous> (/Users/user1/Documents/abp/npm/ng-packs/node_modules/@nrwl/cli/bin/nx.js:43:32)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 47148,
  stdout: null,
  stderr: null
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

## Steps to Reproduce

- Clone the [ABP Framework](https://github.com/abpframework/abp) repo
- Go to the `npm/ng-packs` folder
- Install the packages via yarn command
- Run the `yarn nx migrate latest` command
- Run the `yarn nx migrate --run-migrations` command

**Comment by madipta** [comment]
i have similar issue  when running nx migrate --run-migrations 

> Running migration migration-v12
>     Could not find TypeScript project for project: admin-e2e
> directoryExists is not a function

then the migrations stopped
i figured that because i don't have @angular/cdk installed

after cdk installed the migrations run with error 

> Running migration migration-v12
>   Could not find TypeScript project for project: admin-e2e
>   Could not find TypeScript project for project: others
> Running migration migration-v13
>   Could not find TypeScript project for project: admin-e2e
>   Could not find TypeScript project for project: others

but the migrations finished and my code run like before

**Comment by Coly010** [comment]
@madipta The following output
```
Running migration migration-v12
Could not find TypeScript project for project: admin-e2e
Could not find TypeScript project for project: others
Running migration migration-v13
Could not find TypeScript project for project: admin-e2e
Could not find TypeScript project for project: others
```

Is from the Angular CDK. It's. a poor output, but essentially it just means they're skipping projects they don't need to process, or can't figure out how to process.

You can see more on that on this issue: https://github.com/angular/components/issues/23930

There's a different issue that needs to be addressed here though, and it is being worked on :)

**Comment by leosvelperez** [comment]
Thanks for reporting this!

We are looking at the best way to handle the issue and we'll push a fix for it once we have it.

@mehmet-erim @madipta In your specific use case, to unblock yourself, the right solution is to remove the migrations with `"package": "@angular/cdk"` from the `migrations.json` file, and then, you can run `nx migration --run-migrations`.

**Comment by dbvcode** [comment]
Same issue here. @leosvelperez your hack does indeed work, but won't something be 'lagging' if we skip that migration alone?

**Comment by leosvelperez** [comment]
@dbvcode it's not exactly a hack, but rather the expected behavior. If you don't have a direct dependency on a package, you shouldn't be running a migration for it. The issue here is that the migrator tool is adding the migrations even though you don't directly depend on `@angular/cdk`, it's doing so because one of your dependencies does depend on it. I'll work on a fix to prevent adding migrations for packages that are not direct dependencies of a workspace.

**Comment by westmark** [comment]
@leosvelperez I get roughly the same error message, but when I run `nx migrate latest`, so no `migrations.json` is ever generated. Any tips for me?

```
 > nx migrate latest

/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/yargs.js:1132
                throw err;
                ^

Error: Command failed: yarn
    at checkExecSyncError (node:child_process:707:11)
    at Object.execSync (node:child_process:744:15)
    at taoPath (/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/workspace/src/command-line/nx-commands.js:377:21)
    at Object.handler (/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/workspace/src/command-line/nx-commands.js:113:19)
    at Object.runCommand (/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/command.js:196:48)
    at Object.parseArgs [as _parseArgs] (/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/yargs.js:1043:55)
    at Object.get [as argv] (/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/workspace/node_modules/yargs/build/lib/yargs.js:986:25)
    at Object.initLocal (/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/cli/lib/init-local.js:28:79)
    at Object.<anonymous> (/Users/westmark/projects/reg-n-tech/regtech-applications/node_modules/@nrwl/cli/bin/nx.js:43:18)
    at Module._compile (node:internal/modules/cjs/loader:1109:14) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 92497,
  stdout: null,
  stderr: null
}
```

**Comment by leosvelperez** [comment]
@westmark that's a separate issue, it doesn't happen when running migrations but rather at the start, when the migrations are going to be collected.

Try running `NX_VERBOSE_LOGGING=true nx migrate latest` and see the logs, that might print more information on the actual issue.

**Comment by westmark** [comment]
Sadly, I get zero additional output. But I found #5672, which seems to be the same issue. I'll follow that instead
